src/mapgen/cavegen.h

```diff
@@ -20,6 +20,7 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 
 #pragma once
 
+#include "mapgen.h"
```

Comment:
It is not necessary to increase the size of this header (which is what `#include` effectively does).
Add `class BiomeGen;` after L28. This works because it's pointers (or references).